plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/transport/simulate/SimulateExecutionService.java

```diff
@@ -43,7 +44,7 @@ SimulateDocumentResult executeDocument(Pipeline pipeline, IngestDocument ingestD
             List<SimulateProcessorResult> processorResultList = new ArrayList<>();
             IngestDocument currentIngestDocument = new IngestDocument(ingestDocument);
             for (int i = 0; i < pipeline.getProcessors().size(); i++) {
-                Processor processor = pipeline.getProcessors().get(i);
+                HandledProcessor processor = pipeline.getProcessors().get(i);
```

Comment:
Right now we're not simulating pipeline failures. Maybe we should at some point? But that is outside the scope of this PR.